### PR TITLE
Update terraform.py

### LINKED ISSUE #986: Upgrade to newer version of terraform.py
There are a couple issues in Mantl that could be resolved by upgrading to the latest terraform.py. 

See #768, #985

### PR DESCRIPTION
**Pull request by siddharthist**
Fixes #986

### CODE REVIEW

**Comment by sehqlr** [comment]
I think that by v1.0, we should have terraform.py as a submodule, so upkeep is simpler.

**Comment by siddharthist** [comment]
@sehqlr Agreed.

**Comment by sehqlr** [comment]
This passed a basic install + healthy test. Are there any specific tests for this one?

**Comment by siddharthist** [comment]
@sehqlr Nope! This should be ready to merge after we release 0.6.

File: plugins/inventory/terraform.py [hunk 1]
```diff
@@ -211,13 +212,14 @@ def softlayer_host(resource, module_name):
         'provider': 'softlayer',
     }
 
-    # attrs specific to microservices-infrastructure
+    # attrs specific to Mantl
     attrs.update({
         'consul_dc': _clean_dc(attrs['metadata'].get('dc', attrs['region'])),
-        'role': attrs['metadata'].get('role', 'none')
+        'role': attrs['metadata'].get('role', 'none'),
+        'ansible_python_interpreter': attrs['metadata'].get('python_bin','python')
```
Comment (avnik) [inline plugins/inventory/terraform.py]: Why we need to touch stuff related to python_interpreter here?

Comment (siddharthist) [inline plugins/inventory/terraform.py]: I'm not sure, maybe @BrianHicks would know.

**Comment by sehqlr** [comment]
When we move this to a submodule, where would the clone go? In the root directory?

**Comment by siddharthist** [comment]
@sehqlr Let's make an issue for that and discuss there.

**Comment by sehqlr** [comment]
I made an issue: #1120

**Comment by siddharthist** [comment]
Failure on AWS was quota-related. On GCE, Traefik health checks are failing, but that doesn't seem related. DO build passed.

**Pull request merged**